### DISCUSSION TITLE
IOS app build prompts for login keychain password even if using a temporary keychain

### DISCUSSION THREAD

**Discussion by singhravi1** [fastlane/fastlane#18175] [Q&A]
Hi,

I'm using fastlane with Jenkins and do not want to have any prompt while a build gets generated and pushed to the testflight.
I'm creating temp keychain and unlocking it so that no prompt is required for the keychain access but its still showing a prompt for login keychain password.

I checked [this](https://github.com/fastlane/fastlane/issues/10589#issuecomment-375430190) solution as well but still no change.

The prompt comes after this in the console,
```
▸ Running script '[CP] Copy Pods Resources'
```

My Fastfile look like this,
```
def ensure_keychain(name)
  delete_keychain(
    name: name
  ) if File.exist? File.expand_path("~/Library/Keychains/#{name}-db")
  create_keychain(
    name: name,
    password: 'anytmppass',
    unlock: true,
    timeout: false
  )
end

default_platform(:ios)

platform :ios do
  desc "Push a new build to TestFlight"
  lane :development do
    begin
      ensure_keychain('fastlanep');
      get_push_certificate
      match(type: "appstore", readonly: true, keychain_name: "fastlanep", keychain_password: "anytmppass")
      ensure_git_status_clean
      increment_build_number(xcodeproj: "Project.xcodeproj")
      commit_version_bump(xcodeproj: "Project.xcodeproj")
      push_to_git_remote
      build_app(
          workspace: "Project.xcworkspace",
          scheme: "Project",
          clean: true,
          export_method: "app-store",
          build_path: "./builds",
          output_directory: "./builds"
        )
      upload_to_testflight
      slack(
          message: "Yay!",
          pretext: "iOS Build has been successfully submitted to TestFlight",
          slack_url: "https://hooks.slack.com/services/xxxxx",
          username: "Fastlane",
          success: true,
          fail_on_error: false
        )
    rescue => ex
      slack(
        message: ex.to_s,
        pretext: "Error while uploading the build to testflight",
        slack_url: "https://hooks.slack.com/services/xxxxx",
        username: "Fastlane",
        success: false
      )
      UI.error(ex)
    end
  end
end
```

Has anyone successfully handled this? Any suggestion would be great. Thanks!

**Comment by dinsen** [comment]
Hi @singhravi1 
Have you tried adding some of this parameters?
`default_keychain: true, lock_when_sleeps: false, add_to_search_list: true`

I can see we use them, but we don't delete the keychain first, we keep it. (We have our own Mac mini at the office)

**Reply by singhravi1** [reply]
Hi @Dinsen, thank you for the suggestion.
I tried using, `default_keychain` but it still asked for login keychain password.
I have disabled sleep and lock on my mac so i think `lock_when_sleeps` would not have any affect.
Also, I checked in the doc and the default value of `add_to_search_list` is `true`, so there is no need to specify it.

Can you please suggest how different is your Fastfile then mine? Also how are you managing the CI?

Thanks again!

**Reply by dinsen** [reply]
We don't use Fastfiles, so it's not that easy for me to compare with your setup.
We have build our own CLI that will call into fastlane when needed.

On our Mac mini that is running our CI setup, we check if the keychain exists. If not, we create it with:
```
create_keychain(
  name: "fastlane",
  password: "superSecurePassword",
  default_keychain: true,
  unlock: true,
  timeout: false,
  lock_when_sleeps: false,
  add_to_search_list: true 
)
```

After that we unlock the keychain with this:
`security unlock-keychain -p superSecurePassword ~/Library/Keychains/fastlane-db`

If the keychain exists we still call `security unlock-keychain..`
When using `Match` we have specified `MATCH_KEYCHAIN_NAME` and `MATCH_KEYCHAIN_PASSWORD` environment.

**Reply by singhravi1** [reply]
Oh Okay.
We're not unlocking keychain after creating it as we're already passing unlock parameter while creating. But I'll try adding unlock step.
keychain name is already being passed to the match command, lets add password as well.
I'll give these updates a try and lets see how it goes. Will update here about that.
Thanks!

**Reply by singhravi1** [reply]
This did it. Thanks @Dinsen !

**Reply by dinsen** [reply]
Great! 🎉
